### PR TITLE
MB-2423: Fix android gradle builds

### PR DESCRIPTION
**Pull request by rlepinski**
When Unity uses gradle build, it will add every .aar as a dependency on the main app. In library project gets its own gradle module and must be compiled. If that module tries to access resources outside of its package or dependencies, everything fails. We were having that issue with our unity-plugin accessing resources in the SDK. 

This change now leverages Google Jar Resolver to handle installing our plugin as well as the SDK. It will auto resolve manifest place holders, explode the aar when not using gradle, etc...

However we still need a way to dump resources into the main application for things like our notification icon and airship config. So now I have a very lightweight "urbanairship-resources" projects that can be used as a resource dump.

I was able to  build with both android build systems.

### CODE REVIEW

File: unity-plugin/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -2,7 +2,6 @@
 <manifest
     package="com.urbanairship.unityplugin"
     xmlns:android="http://schemas.android.com/apk/res/android">
-    <uses-sdk android:minSdkVersion="16"/>
```
Comment (marc-scig) [inline unity-plugin/src/main/AndroidManifest.xml]: Is this still needed for the plugin? I see that it's in the resources manifest, but I'd assume we'd want that here as well.

Comment (rlepinski) [reply unity-plugin/src/main/AndroidManifest.xml]: Its only needed for library style builds. AAR it replaces it.

**Comment by crow** [comment]
LGTM 🍔

**Comment by marc-scig** [comment]
⚡️

**Pull request merged**